src/plugins/vis_type_timeseries/common/panel_types.ts

```diff
@@ -25,3 +27,5 @@ export const PANEL_TYPES = {
   TIMESERIES: 'timeseries',
   METRIC: 'metric',
 };
+
+export type PANEL_TYPES = $Values<typeof PANEL_TYPES>;
```

Comment:
just interesting, maybe PANEL_TYPES should be enum?

Reply:
yes it could be
but in such a case I'd also need to update other parts of code, since in some cases we are trying to access a property of `PANEL_TYPES` using string type, e.x. :

![image](https://user-images.githubusercontent.com/31325372/99675561-69919700-2a88-11eb-9126-0c4203fb2626.png)


where `panel.type` should be a type of `PANEL_TYPES`, but it is a string since it comes from schema object.
I could use type casting here, but don't think it's necessary in scope of this pr..

Reply:
what if we do: 
1) src/plugins/vis_type_timeseries/common/ui_restrictions.ts
```
export const limitOfSeries: Partial<Record<PANEL_TYPES, number>> = {
  [PANEL_TYPES.GAUGE]: 1,
  [PANEL_TYPES.METRIC]: 2,
};
```
2) src/plugins/vis_type_timeseries/common/vis_schema.ts 
```
type: schema.oneOf([
    schema.literal('table'),
    schema.literal('gauge'),
    schema.literal('markdown'),
    schema.literal('top_n'),
    schema.literal('timeseries'),
    schema.literal('metric'),
  ]),
```

Reply:
ok, done.